oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/remote/RemoteServiceHandler.java

```diff
@@ -117,24 +117,33 @@ public void onNext(RemoteMessage message) {
                     String nextWorkerName = message.getNextWorkerName();
                     RemoteData remoteData = message.getRemoteData();
 
-                    try {
-                        RemoteHandleWorker handleWorker = workerInstanceGetter.get(nextWorkerName);
-                        if (handleWorker != null) {
-                            AbstractWorker nextWorker = handleWorker.getWorker();
-                            StreamData streamData = handleWorker.getStreamDataClass().newInstance();
+                    RemoteHandleWorker handleWorker = workerInstanceGetter.get(nextWorkerName);
+                    if (handleWorker != null) {
+                        AbstractWorker nextWorker = handleWorker.getWorker();
+                        StreamData streamData;
+                        try {
+                            streamData = handleWorker.getStreamDataClass().newInstance();
+                        } catch (Throwable t) {
+                            remoteInErrorCounter.inc();
+                            LOGGER.error(t.getMessage(), t);
+                            return;
+                        }
+                        try {
                             streamData.deserialize(remoteData);
-                            nextWorker.in(streamData);
-                        } else {
-                            remoteInTargetNotFoundCounter.inc();
-                            LOGGER.warn(
-                                "Work name [{}] not found. Check OAL script, make sure they are same in the whole cluster.",
-                                nextWorkerName
-                            );
+                        } catch (Throwable t) {
+                            remoteInErrorCounter.inc();
+                            LOGGER.error("Can't deserialize data {}, this data is discarded.", message, t);
+                            return;
                         }
-                    } catch (Throwable t) {
-                        remoteInErrorCounter.inc();
-                        LOGGER.error(t.getMessage(), t);
+                        nextWorker.in(streamData);
+                    } else {
+                        remoteInTargetNotFoundCounter.inc();
+                        LOGGER.warn(
+                            "Data is discarded due to worker not found. Check OAL/MAL script, make sure they are aligned in the whole cluster. The data is {}",
+                            message
```

Comment:
I think this is unhelpful. Message doesn't have a readable toString representation

Reply:
This is a proto object. Isn't it output the data? I read this from decompiling codes.

```
  @Override
  public final String toString() {
    return TextFormat.printer().printToString(this);
  }
```

Reply:
For safety, I run the demo codes locally
```java
public static void main(String[] args) {
        RemoteMessage obj = RemoteMessage.newBuilder()
                                             .setNextWorkerName("dbc").build();
        System.out.println("Default toString(): " + obj.toString());
        System.out.println("TextFormat: " + TextFormat.printer().printToString(obj));

        try {
            System.out.println("JsonFormat: " + JsonFormat.printer().print(obj));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
```

```
Default toString(): nextWorkerName: "dbc"

TextFormat: nextWorkerName: "dbc"

JsonFormat: {
  "nextWorkerName": "dbc"
}
```